commands/core/site_install.drush.inc

```diff
@@ -289,4 +300,13 @@ function drush_core_site_install($profile = NULL) {
   }
   drush_include_engine('drupal', 'site_install');
   drush_core_site_install_version($profile, $form_options);
+
+  // Post installation run the configuration import.
+  if ($config = drush_get_option('config')) {
+    // Set the destination site UUID to match the source UUID, to bypass a core fail-safe.
+    $source_storage = new FileStorage($config);
+    drush_op('drush_config_set', 'system.site', 'uuid', $source_storage->read('system.site')['uuid']);
```

Comment:
No, it modifies the UUID in the active config of the newly installed site. I've been using @markdorison's variation on this and it has never modified the source yml file on disk.